product-service/handlers/product-item.ts

```diff
@@ -6,38 +6,275 @@ import {
 import 'source-map-support/register';
 import { throwError } from '../helpers/error.helper';
 import { ErrorsEnum } from '../types/errors.enum';
-import data from '../fake-data/MOCK_DATA.json';
-import { IProductItem } from '../types/product-item.interface';
 import {
     formattedErrorResponse,
     formattedSuccessResponse,
 } from '../helpers/response.helper';
+import { dbOptions } from '../helpers/db.helper';
+import * as pg from 'pg';
+import { isUUIDv4 } from '../helpers/request.helper';
 
 export const getProductsById: APIGatewayProxyHandler = async (
     event: APIGatewayProxyEventBase<APIGatewayEventDefaultAuthorizerContext>
 ) => {
+    console.log('📝 getProductsById: ', event);
+
+    const client = new pg.Client(dbOptions);
+
+    try {
+        const id = event?.pathParameters?.productId;
+
+        if (id == null || (id && !isUUIDv4(id))) {
+            throwError(ErrorsEnum.WrongRequest, 400);
+        }
+
+        await client
+            .connect()
+            .catch((dbErr) =>
+                throwError(ErrorsEnum.DBError, 500, JSON.stringify(dbErr))
+            );
+
+        const { rows: products } = await client
+            .query(
+                `select
+                     products.id,
+                     products.title,
+                     products.description,
+                     products.price,
+                     stocks.count
+                 from
+                     products
+                 left join stocks on
+                     products.id = stocks.product_id
+                 where products.id = '${id}'`
+            )
+            .catch((dbErr) =>
+                throwError(ErrorsEnum.DBError, 500, JSON.stringify(dbErr))
+            );
+
+        if (!products[0]) {
+            throwError(ErrorsEnum.NotFoundData, 404);
+        }
+
+        return formattedSuccessResponse(products[0]);
+    } catch (error) {
+        return formattedErrorResponse(error);
+    } finally {
+        client.end();
+    }
+};
+
+export const createProducts: APIGatewayProxyHandler = async (
+    event: APIGatewayProxyEventBase<APIGatewayEventDefaultAuthorizerContext>
+) => {
+    console.log('📝 createProducts: ', event);
+
+    const client = new pg.Client(dbOptions);
+
     try {
-        const id =
-            event &&
-            event.pathParameters &&
-            Number(event.pathParameters.productId);
+        const data = JSON.parse(event?.body || null);
 
-        if (id == null) {
+        if (
+            data == null ||
+            !data?.title ||
+            !data?.description ||
+            !data?.price ||
+            !data?.count
```

Comment:
Maybe add checks not only on existance, but also on type?

Reply:
Wel, thxl :) I would say it's not the main goal of this task & course. Of course, it's nice to have. But...

Reply:
Yeah, it is optional)